--- _effect-chunk.mjs.orig	2026-09-21 17:13:38.356694868 +0200
+++ _effect-chunk.mjs	2026-09-21 17:13:38.376694872 +0200
@@ -40,6 +40,22 @@
   consumerMarkedDirty: () => {},
   consumerOnSignalRead: () => {}
 };
+function isValidLink(checkLink, consumer) {
+  const producersTail = consumer.producersTail;
+  if (producersTail !== undefined) {
+    let link = consumer.producers;
+    do {
+      if (link === checkLink) {
+        return true;
+      }
+      if (link === producersTail) {
+        break;
+      }
+      link = link.nextProducer;
+    } while (link !== undefined);
+  }
+  return false;
+}
 function producerAccessed(node) {
   if (inNotificationPhase) {
     throw new Error(typeof ngDevMode !== 'undefined' && ngDevMode ? `Assertion error: signal read during notification phase` : '');
@@ -64,7 +80,7 @@
     }
   }
   const prevConsumerLink = node.consumersTail;
-  if (prevConsumerLink !== undefined && prevConsumerLink.consumer === activeConsumer && (!isRecomputing || prevConsumerLink.knownValidAtEpoch === epoch)) {
+  if (prevConsumerLink !== undefined && prevConsumerLink.consumer === activeConsumer && (!isRecomputing || isValidLink(prevConsumerLink, activeConsumer))) {
     return;
   }
   const isLive = consumerIsLive(activeConsumer);
